console-csharp-connect-sample/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Graph;
+
+namespace console_csharp_connect_sample.Helpers
+{
+	// This class encapsulates the details of getting a token from MSAL and exposes it via the 
+	// IAuthenticationProvider interface so that GraphServiceClient or AuthHandler can use it.
+	// A significantly enhanced version of this class will in the future be available from
+	// the GraphSDK team. It will support all the types of Client Application as defined by MSAL.
+	public class MsalAuthenticationProvider : IAuthenticationProvider
+	{		
+		private PublicClientApplication _clientApplication;
+		private  string[] _scopes;
+	    private string _userToken = null;
```

Comment:
Please see https://github.com/andrueastman/consoleApp-deltaQuery-dotNet-sample/pull/1
don't keep the user token. this will break the program if it takes too long to run. you need to call AcquireTokenSilentAsync, and if it fails, AcquireTokenSilent

Reply:
Fixed this one. Thanks for the tip @jmprieur.